p2p/p2p.md

```diff
@@ -0,0 +1,220 @@
+# P2P Design
+
+`p2p` package should expose minimal functions as a service to other components in Spacemesh.
+- `SendMessage(MSG, nodeID)`
+- `Broadcast(MSG)`
+- `OnRecv(msgtype, optional:peer, callback(func or chan))`  / `RegisterProtocol(Protocol, handler)`
+
+## Node
+
+The `LocalNode` is the struct that represent a node in our network. in order to use the above P2P functionality a `LocalNode` must be created from within the `p2p` package.
+It holds a Private and Public key and an Identity
+`LocalNode` can initialize from scratch or form existing persistent keys and files. it needs a local address and port.
+
+To enable this it implements multiple components:
+
+## Swarm
+`Swarm` manages the communications with peers while maintaining a list of healthy connected peers and an updated `RoutingTable` of nodes to choose from.
+`Swarm`'s responsibility is to receive and send network messages and connections using events propagated from `Net`, handle them - check that they are in a valid format and time and pass them over to the right protocol handler using `Demuxer`.
```

Comment:
Not sure I understand the meaning of "and connections" in "Swarm's responsibility is to receive and send network messages and connections".